src/LoopInterface.php

```diff
@@ -240,6 +240,42 @@ public function isTimerActive(TimerInterface $timer);
     public function futureTick(callable $listener);
 
     /**
+     * Registers a signal listener with the loop, which
+     * on it's turn registers it with a signal handler
+     * suitable for the loop implementation.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @throws UnsupportedFeatureException when signals
+     * aren't supported by the loop, e.g. when required
+     * extensions are missing.
+     *
+     * @return void
+     */
+    public function addSignal($signal, callable $listener);
+
+    /**
+     * Removed previous registered signal listener from
+     * the loop, which on it's turn removes it from the
+     * underlying signal handler.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @throws UnsupportedFeatureException when signals
+     * aren't supported by the loop, e.g. when required
+     * extensions are missing.
```

Comment:
It is more about consistency between two methods than it is likely someone will call this method when `addSignal` throws.